src/pauli_exp_val.rs

```diff
@@ -80,11 +91,17 @@ pub fn expval_pauli_with_x(
     phase: Complex64,
     x_max: u32,
 ) -> PyResult<f64> {
+    if num_qubits > usize::BITS as usize {
+        return Err(PyOverflowError::new_err(format!(
+            "The value for num_qubits, {}, is too larget and would overflow",
+            num_qubits
+        )));
```

Comment:
A more wordy version in case it's useful.

```rs
format!(
    "The value specified as '{}' for parameter 'num_qubits' must be < '{}' to avoid arithmetic overflow.",
    num_qubits,
    usize::BITS
)
```

Reply:
Resolving this since it's not likely to be hit, and so the current error message is fine IMO.